OSDOCS-17693: adds install to MCP gateway docs (#109364)
openshift/openshift-docs: modules/ref-mcp-gateway-auto-route.adoc

// Module included in the following assemblies:
//
// *mcp_gateway_install/mcp-gateway-install.adoc

:_mod-docs-content-type: REFERENCE
[id="ref-mcp-gateway-auto-route_{context}"]
= Automatically created HTTPRoute example

[role="_abstract"]
Applying the `MCPGatewayExtension` custom resources (CR) generates an automatic `HTTPRoute` CR. The following is an example of the route.

.Example automatically generated HTTPRoute CR
[source,yaml]
----
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: mcp-gateway-route
  namespace: mcp-gateway-namespace
spec:
  parentRefs:
  - name: mcp-gateway
    namespace: istio-system
    sectionName: mcp
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /mcp
    backendRefs:
    - name: mcp-gateway
      port: 8080
----

* The value of the `spec.parentRefs.name:` parameter must match your `Gateway` CR name.
* The value of the `spec.parentRefs.namespace:` parameter must match your `Gateway` namespace.

> * The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.targetRef.name:` value of your listener.

Reviewer: ```suggestion
* The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.targetRef.sectionName:` value of your listener.
```

> * The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.TargetRefs.name:` value of your listener.

Reviewer: ```suggestion
* The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.targetRef.sectionName:` value of your listener.
```

> * The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.targetRef.sectionName:` value of your listener.

Reviewer 2: The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.listeners[*].name` value of your listener

- or something like that. Basically `sectionName` here actually is a name of one of the Gateway's Listeners

// Module included in the following assemblies:
//
// *mcp_gateway_install/mcp-gateway-install.adoc

:_mod-docs-content-type: REFERENCE
[id="ref-mcp-gateway-auto-route_{context}"]
= Automatically created HTTPRoute example

[role="_abstract"]
Applying the `MCPGatewayExtension` custom resources (CR) generates an automatic `HTTPRoute` CR. The following is an example of the route.

.Example automatically generated HTTPRoute CR
[source,yaml]
----
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: mcp-gateway-route
  namespace: mcp-gateway-namespace
spec:
  parentRefs:
  - name: mcp-gateway
    namespace: istio-system
    sectionName: mcp
  rules:
  - matches:
    - path:
        type: PathPrefix

>         value: /

Reviewer: I think this should be /mcp 
https://github.com/Kuadrant/mcp-gateway/blob/d3921a2c0e96d08af303c4a794ba96b2834e40ba/internal/controller/broker_router.go#L519:

    backendRefs:
    - name: mcp-gateway
      port: 8080
----
+

> * The value of the `spec.parentRefs.name:` parameter must match your `Gateway` CR name.

Reviewer 3: ```suggestion
----

* The value of the `spec.parentRefs.name:` parameter must match your `Gateway` CR name.
```

* The value of the `spec.parentRefs.namespace:` parameter must match your `Gateway` namespace.
* The value of the `spec.parentRefs.sectionName:` parameter must match the `spec.TargetRefs.name:` value of your listener.

> * The default value of `\` in the `spec.rules.matches.path.value:` parameter catches all MCP traffic.

Reviewer: same as above
